### ISSUE TITLE
Scripting request: get current system version

### ISSUE THREAD

**Issue by siema** [d0k3/GodMode9#209]
Would it be possible to expose current system version string as ENV? It'd be really helpful in organizing backups :)

**Comment by d0k3** [comment]
I see how this would be helpful, but recognizing system version is not exactly trivial in ARM9 and would also  require me to keep an internal, ever updating list of information to recognize each system version (and, on top, that for all versions, separate for O3DS and N3DS, and separate for devkit and retail).

In short, massive clusterfuck, won't be added.

**Comment by fox8091** [comment]
Not as much of a clusterfuck as you may think. The version is contained
within CVer and NVer (000400DB00016X02 on O3DS for CVer, 000400DB2016X02
for N3DS CVer and 000400DB00017X02 for NVer for all) in a file called
version.bin. CVer contains the X.X.X-YX (Only the Xs) and NVer contains
Y.Y.Y-XY (Only the X). The format of version.bin is documented [here]{
https://www.3dbrew.org/wiki/CVer} for CVer and [here]{
https://www.3dbrew.org/wiki/NVer} for NVer.

On Sep 19, 2017 6:30 AM, "d0k3" <notifications@github.com> wrote:

I see how this would be helpful, but recognizing system version is not
exactly trivial in ARM9 and would also require me to keep an internal, ever
updating list of information to recognize each system version (and, on top,
that for all versions, separate for O3DS and N3DS, and separate for devkit
and retail).

In short, massive clusterfuck, won't be added.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/d0k3/GodMode9/issues/209#issuecomment-330510782>, or mute
the thread

**Comment by d0k3** [comment]
That'S assuming we can properly browse system saves outside of userland, without system functions.

**Comment by ihaveamac** [comment]
I wouldn't imagine it being too hard, given you can already mount NCCH images, you could mount CVer/NVer NCCH/RomFS and read version.bin for the numbers.

**Comment by d0k3** [comment]
I actually mistook cver and nver for system configuration data... well, it would be possible, but for now, still not worth it. Mainly because that would be a lot of code and also a noticable delay before running scripts for a rather minor convenience.

**Comment by siema** [comment]
Didn't expect it to be so obscured in ARM9. Thanks for explaining!

**Comment by ihaveamac** [comment]
titleversion.txt was only added in 9.6.0-X, anything older would still require version.bin parsing. it's not too hard though: https://www.3dbrew.org/wiki/CVer
